tools/clang/tools/dxcompiler/dxcompilerobj.cpp

```diff
@@ -1334,6 +1334,8 @@ class DxcCompiler : public IDxcCompiler3,
     compiler.getDiagnosticOpts().ShowOptionNames = Opts.ShowOptionNames ? 1 : 0;
     compiler.getDiagnosticOpts().Warnings = std::move(Opts.Warnings);
     compiler.getDiagnosticOpts().VerifyDiagnostics = Opts.VerifyDiagnostics;
+    compiler.getDiagnosticOpts().setVerifyIgnoreUnexpected(
+        static_cast<clang::DiagnosticLevelMask>(Opts.DiagMask));
```

Comment:
I don't think we should be using a static cast without guaranteeing somehow that the values are the same. So either we should add `static_assert` to verify the enum values match or add a function to translate between the enums with a switch statement.

Reply:
Added static_assert.

Reply:
Does this actually check that the enum values match? Or is this just checking they have the same underlying type?

Reply:
Updated to check enum values.

Reply:
That looks good, thanks. I think it kind of interrupts the flow a bit to have it inline with where we set the value. Might want to move it outside the function.